SerialPrograms/Source/PokemonLA/Programs/ShinyHunting/PokemonLA_BurmyFinder.cpp

```diff
@@ -268,6 +268,182 @@ void BurmyFinder::go_to_height_camp(SingleSwitchProgramEnvironment& env, BotBase
     }
 }
 
+size_t BurmyFinder::grouped_path(SingleSwitchProgramEnvironment& env, BotBaseContext& context, size_t path, TreeCounter& tree_counter){
+
+    size_t last_checked_tree = 0;
+
+    if(path != 0)
+        go_to_height_camp(env, context);
+
+    env.console.log("Currently Checking Path:" + std::to_string(path));
+
+    BattleMenuDetector battle_menu_detector(env.console, env.console, true);
+
+    int ret = run_until(
+        env.console, context,
+        [&](BotBaseContext& context){
+            switch (path) {
+            case 0:
+                // === Fly to the first tree on trip ===
+                env.console.log("Checking tree: 0");
+                pbf_move_left_joystick(context, 255, 90, 20, (0.5 * TICKS_PER_SECOND));
+                pbf_press_button(context, BUTTON_ZL, 20, (0.5 * TICKS_PER_SECOND));
+                change_mount(env.console, context, MountState::BRAVIARY_ON);
+                pbf_press_button(context, BUTTON_B, (5 * TICKS_PER_SECOND), 0);
+                pbf_press_button(context, BUTTON_Y, (4.2 * TICKS_PER_SECOND), 0);
+                pbf_press_button(context, BUTTON_PLUS, 20, (1 * TICKS_PER_SECOND));
+
+                // Now on ground, move towards the tree
+                // change_mount(env.console, context, MountState::WYRDEER_OFF);
+                pbf_move_left_joystick(context, 0, 70, 20, (0.3 * TICKS_PER_SECOND));
+                // Change camera to face what player character faces
+                pbf_press_button(context, BUTTON_ZL, 20, (0.4 * TICKS_PER_SECOND));
+                // Move camera down
+                pbf_move_right_joystick(context, 128, 255, (0.3 * TICKS_PER_SECOND), (0.2 * TICKS_PER_SECOND));
+
+                // Now we should be on ground
+                disable_shiny_sound(context);
+                check_tree_no_stop(env, context);
+                context.wait_for_all_requests();
+                last_checked_tree = 0;
+
+                // === Fly to the second tree on trip ===
+                // Mount on Braviary
+                env.console.log("Checking tree: 1");
+                pbf_press_button(context, BUTTON_PLUS, 20, 100);
+                enable_shiny_sound(context);
+
+                // Now face towards the next tree
+                pbf_move_left_joystick(context, 255, 200, 1 * TICKS_PER_SECOND, 0);
+                pbf_mash_button(context, BUTTON_B, 100);
+                pbf_press_button(context, BUTTON_Y, (2.1 * TICKS_PER_SECOND), 0);
+                // Get off Braviary
+                pbf_press_button(context, BUTTON_PLUS, 20, (1 * TICKS_PER_SECOND));
+                // Move camera down
+                pbf_move_right_joystick(context, 128, 255, (0.1 * TICKS_PER_SECOND), (0.3 * TICKS_PER_SECOND));
+
+                // Now we should be on ground
+                disable_shiny_sound(context);
+                check_tree_no_stop(env, context);
+                context.wait_for_all_requests();
```

Comment:
Battle menu detector might find a battle during the wait here. But `last_checked_tree` is still 1. What about moving the assignment of `last_checked_tree` before `check_tree_no_stop()`?
Same to other places where `last_checked_tree` is changed.

Reply:
good point, changed!